File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int bowl = arr[0];
        Queue queue = new LinkedList();
        
        queue.offer(arr[0]);
        for(int i = 1; i < arr.length; i++){
            if(bowl != arr[i]){
                queue.offer(arr[i]);
            }
            bowl = arr[i];
        }
        
        int[] answer = new int[queue.size()];
        int cnt = 0;
        while(queue.size() > 0){
            answer[cnt] = (int)queue.poll();
            cnt++;
        }
            
        
        return answer;
    }
}

File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr){
    var answer = [];
    let trust = true;
    for(let i = 0; i < arr.length - 1; i++){
        if(arr[i] != arr[i+1]){
            answer.push(arr[i]);
            trust = true;
        }else{
            trust = false;
        }
    }
    answer.push(arr[arr.length - 1]);
    
    return answer;
}